### ISSUE TITLE
migrate-to-repokey not accepting valid passphrase in borg 1.2.0

### ISSUE THREAD

**Issue by Alives** [borgbackup/borg#7001]
## Have you checked borgbackup docs, FAQ, and open Github issues?
Yes

## Is this a BUG / ISSUE report or a QUESTION?
Issue

## System information. For client/server mode post info for both machines.
amd64, backing up and writing to ext4

#### Your borg version (borg -V).
`Borg: 1.2.0  Python: CPython 3.10.4 msgpack: 1.0.3 fuse: pyfuse3 3.2.0 [pyfuse3,llfuse]`

#### Operating system (distribution) and version.
`Ubuntu Jammy 22.04.1 LTS amd64`

#### Hardware / network configuration, and filesystems used.
ext4, WD 10TB sata WDC WD100EMAZ-00WJTA0 disk

#### How much data is handled by borg?
5.5TB

#### Full borg commandline that lead to the problem (leave away excludes and passwords)
`borg key migrate-to-repokey /data/backup/server`

## Describe the problem you're observing.
```
# borg key migrate-to-repokey /data/backup/server
Enter passphrase for /data/backup/server:
Enter passphrase for /data/backup/server:
Enter passphrase for /data/backup/server:
exceeded the maximum password retries
```

However, that passphrase is fine:
```
# borg info /data/backup/server
Enter passphrase for key /root/.config/borg/keys/server.borgbackup.key:
Repository ID: 08ffbcc03cec5a53f495082710ea4c52603af50053485c985dd59a02aa6a78d0
Location: /data/backup/server
Encrypted: Yes (key file)
Key file: /root/.config/borg/keys/server.borgbackup.key
Cache: /root/.cache/borg/08ffbcc03cec5a53f495082710ea4c52603af50053485c985dd59a02aa6a78d0
Security dir: /root/.config/borg/security/08ffbcc03cec5a53f495082710ea4c52603af50053485c985dd59a02aa6a78d0
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:                5.50 TB              5.20 TB            216.36 GB

                       Unique chunks         Total chunks
Chunk index:                 1028662             33306887
```

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.
If I specify the passphrase as the `BORG_PASSPHRASE` env var, I'm only prompted for it twice which leads me to believe some portion of this workflow is working with the passphrase.  If I don't use the `BORG_PASSPHRASE` env var, then I'm prompted 3x.

#### Include any warning/errors/backtraces from the system logs
```
borg --debug key migrate-to-repokey /data/backup/server
using builtin fallback logging configuration
38 self tests completed in 0.06 seconds
Verified integrity of /data/backup/server/index.951318
Enter passphrase for /data/backup/server:
Enter passphrase for /data/backup/server:
Enter passphrase for /data/backup/server:
exceeded the maximum password retries
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 5089, in main
    exit_code = archiver.run(args)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 5020, in run
    return set_ec(func(args))
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 183, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 407, in do_migrate_to_repokey
    key_old = PassphraseKey.detect(repository, manifest_data)
  File "/usr/lib/python3/dist-packages/borg/crypto/key.py", line 567, in detect
    raise PasswordRetriesExceeded
borg.crypto.key.PasswordRetriesExceeded: exceeded the maximum password retries

Platform: Linux server 5.15.0-47-generic #51-Ubuntu SMP Thu Aug 11 07:51:15 UTC 2022 x86_64
Linux: Unknown Linux
Borg: 1.2.0  Python: CPython 3.10.4 msgpack: 1.0.3 fuse: pyfuse3 3.2.0 [pyfuse3,llfuse]
PID: 2347893  CWD: /root
sys.argv: ['/usr/bin/borg', '--debug', 'key', 'migrate-to-repokey', '/data/backup/server']
SSH_ORIGINAL_COMMAND: None
```

**Comment by ThomasWaldmann** [comment]
Hmm, that looks like a bug.

Can you reproduce this with a fresh repo? Can you reproduce this with a pure ASCII passphrase?

**Comment by Alives** [comment]
New repo with the same passphrase as my problematic repo has the same issue:
```
# borg init --encryption=keyfile /tmp/test
Enter new passphrase:
Enter same passphrase again:
Do you want your passphrase to be displayed for verification? [yN]: n

By default repositories initialized with this version will produce security
errors if written to with an older version (up to and including Borg 1.0.8).

If you want to use these older versions, you can disable the check by running:
borg upgrade --disable-tam /tmp/test

See https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-0-9-manifest-spoofing-vulnerability for details about the security implications.

IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!
If you used a repokey mode, the key is stored in the repo, but you should back it up separately.
Use "borg key export" to export the key, optionally in printable format.
Write down the passphrase. Store both at safe place(s).
```

```
# borg info /tmp/test
Enter passphrase for key /root/.config/borg/keys/tmp_test:
Repository ID: 3d9ef7c50cda9146c77072410314bdfab571b400d834da2d51546068d0d94ab3
Location: /tmp/test
Encrypted: Yes (key file)
Key file: /root/.config/borg/keys/tmp_test
Cache: /root/.cache/borg/3d9ef7c50cda9146c77072410314bdfab571b400d834da2d51546068d0d94ab3
Security dir: /root/.config/borg/security/3d9ef7c50cda9146c77072410314bdfab571b400d834da2d51546068d0d94ab3
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:                    0 B                  0 B                  0 B

                       Unique chunks         Total chunks
Chunk index:                       0                    0
```

```
# borg key migrate-to-repokey /tmp/test
Enter passphrase for /tmp/test:
Enter passphrase for /tmp/test:
Enter passphrase for /tmp/test:
exceeded the maximum password retries
```

**Comment by Alives** [comment]
Same problem using a passphrase of `passphrase`:
```
# borg init --encryption=keyfile /tmp/test
Enter new passphrase:
Enter same passphrase again:
Do you want your passphrase to be displayed for verification? [yN]: y
Your passphrase (between double-quotes): "passphrase"
Make sure the passphrase displayed above is exactly what you wanted.

By default repositories initialized with this version will produce security
errors if written to with an older version (up to and including Borg 1.0.8).

If you want to use these older versions, you can disable the check by running:
borg upgrade --disable-tam /tmp/test

See https://borgbackup.readthedocs.io/en/stable/changes.html#pre-1-0-9-manifest-spoofing-vulnerability for details about the security implications.

IMPORTANT: you will need both KEY AND PASSPHRASE to access this repo!
If you used a repokey mode, the key is stored in the repo, but you should back it up separately.
Use "borg key export" to export the key, optionally in printable format.
Write down the passphrase. Store both at safe place(s).
```

```
# borg info /tmp/test
Enter passphrase for key /root/.config/borg/keys/tmp_test:
Repository ID: bc8465e01c94320ad85e500946832a341e55e92caf687c4a4e7b98363ba6b835
Location: /tmp/test
Encrypted: Yes (key file)
Key file: /root/.config/borg/keys/tmp_test
Cache: /root/.cache/borg/bc8465e01c94320ad85e500946832a341e55e92caf687c4a4e7b98363ba6b835
Security dir: /root/.config/borg/security/bc8465e01c94320ad85e500946832a341e55e92caf687c4a4e7b98363ba6b835
------------------------------------------------------------------------------
                       Original size      Compressed size    Deduplicated size
All archives:                    0 B                  0 B                  0 B

                       Unique chunks         Total chunks
Chunk index:                       0                    0
```

```
# borg key migrate-to-repokey /tmp/test
Enter passphrase for /tmp/test:
Enter passphrase for /tmp/test:
Enter passphrase for /tmp/test:
exceeded the maximum password retries
```

**Comment by ThomasWaldmann** [comment]
```
        This command migrates a repository from passphrase mode (removed in Borg 1.0)
        to repokey mode.

        You will be first asked for the repository passphrase (to open it in passphrase
        mode). This is the same passphrase as you used to use for this repo before 1.0.

        It will then derive the different secrets from this passphrase.

        Then you will be asked for a new passphrase (twice, for safety). This
        passphrase will be used to protect the repokey (which contains these same
        secrets in encrypted form). You may use the same passphrase as you used to
        use, but you may also use a different one.

        After migrating to repokey mode, you can change the passphrase at any time.
        But please note: the secrets will always stay the same and they could always
        be derived from your (old) passphrase-mode passphrase.
```

Sorry that I did not notice this more quickly: this likely is not the correct command. You only need it for very old repos, see description above (the "passphrase mode" mentioned there is something different than "keyfile" mode).

What you likely need is `borg key change-location` , but that is not yet available in a released borg version (will likely come in borg 2.0).

https://borgbackup.readthedocs.io/en/2.0.0b1/usage/key.html
